File: python-api/src/api/handlers/polls/createPoll/createPollHandler.py
try:
    import unzip_requirements
except ImportError:
    pass

import boto3
import os
import simplejson as json
import uuid

dynamodb_client = boto3.resource('dynamodb')
TABLE_NAME = os.environ['POLLS_TABLE_NAME']
DYNAMO_POLLS_ID = os.environ['POLLS_ID']

def create_response(code, message):
  return {
    'statusCode': code,
    'headers': {'Content-Type': 'application/json'},
    'body': json.dumps(message)
  }

def createPoll(event, context):

    request = json.loads(event['body'])
    poll_id = str(uuid.uuid4())

    params = {
      'Item': {
        'PK': DYNAMO_POLLS_ID + '#' + poll_id,
        'SK': DYNAMO_POLLS_ID + '#' + poll_id,
        'GSI1PK': 'POLLS',
        'title': request['title'],
        'description': request['description'],
        'groups': request['groups'],
      }
    }

    print('params', params)

    try:
      table = dynamodb_client.Table(TABLE_NAME)
      response = table.put_item(**params)
    except Exception as ex:
      return create_response(500, str(ex))


    print('Response', response)

    if response['ResponseMetadata']['HTTPStatusCode'] != 200:
      return create_response(500, 'Error creating the poll')

    return create_response(200, params['Item'])

File: nodejs-api/src/api/handlers/polls/createPoll/createPollHandler.js
'use strict';

const AWS = require('aws-sdk')
const db = new AWS.DynamoDB.DocumentClient({})
const { v4: uuidv4 } = require('uuid')

const tableName = process.env.POLLS_TABLE_NAME
const dynamoPollsId = process.env.POLLS_ID

const response = (statusCode, message) => {
    return {
        statusCode: statusCode,
        body: JSON.stringify(message)
    }
}

const createPoll = async (event) => {
    const reqBody = JSON.parse(event.body)

    const pollId = uuidv4()
  
    const poll = {
      PK: dynamoPollsId + '#' + pollId,
      SK: dynamoPollsId + '#' + pollId,
      GSI1PK: 'POLLS',
      title: reqBody.title,
      description: reqBody.description,
      groups: reqBody.groups
    }

    const params = {
      TableName: tableName,
      Item: poll
    }

    try {
      const createPoll = await db.put(params).promise()
      console.log('createPoll: ', createPoll)

      return response(200, poll)
    } catch (err) {
      console.log('err: ', err)
      return response(500, 'Error creating the Poll')
    }
}

module.exports = {
    createPoll
}